static/app/views/organizationCreate/index.tsx

```diff
@@ -1,75 +1,54 @@
 import NarrowLayout from 'sentry/components/narrowLayout';
 import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {t, tct} from 'sentry/locale';
 import ConfigStore from 'sentry/stores/configStore';
 import {OrganizationSummary} from 'sentry/types';
+import {
+  getRegionChoices,
+  shouldDisplayRegions,
+} from 'sentry/utils/regions';
+import useApi from 'sentry/utils/useApi';
 import {normalizeUrl} from 'sentry/utils/withDomainRequired';
 
-enum RegionDisplayName {
-  US = '🇺🇸 United States of America (US)',
-  DE = '🇪🇺 European Union (EU)',
-}
-
-function getRegionChoices(): [string, string][] {
-  const regions = ConfigStore.get('regions') ?? [];
-
-  return regions.map(({name, url}) => {
-    const regionName = name.toUpperCase();
-    if (RegionDisplayName[regionName]) {
-      return [url, RegionDisplayName[regionName]];
-    }
-
-    return [url, name];
-  });
-}
-
-function getDefaultRegionChoice(
-  regionChoices: [string, string][]
-): [string, string] | undefined {
-  if (!shouldDisplayRegions()) {
-    return undefined;
-  }
-
-  const usRegion = regionChoices.find(
-    ([_, regionName]) => regionName === RegionDisplayName.US
-  );
-
-  if (usRegion) {
-    return usRegion;
-  }
-
-  return regionChoices[0];
-}
-
-function shouldDisplayRegions(): boolean {
-  const regionCount = (ConfigStore.get('regions') ?? []).length;
-  return (
-    ConfigStore.get('features').has('organizations:multi-region-selector') &&
-    regionCount > 1
-  );
-}
-
-function removeRegionFromRequestForm(formData: Record<string, any>) {
-  const shallowFormDataCopy = {...formData};
-
-  delete shallowFormDataCopy.region;
-  return shallowFormDataCopy;
+function removeDataStorageLocationFromFormData(
+  formData: Record<string, any>
+): Record<string, any> {
+  const shallowFormDataClone = {...formData};
+  delete shallowFormDataClone.dataStorageLocation;
+  return shallowFormDataClone;
 }
 
 function OrganizationCreate() {
   const termsUrl = ConfigStore.get('termsUrl');
   const privacyUrl = ConfigStore.get('privacyUrl');
   const regionChoices = getRegionChoices();
-  const [regionUrl, setRegion] = useState<string | undefined>(
-    getDefaultRegionChoice(regionChoices)?.[0]
-  );
+  const client = useApi();
+
+  // This is a trimmed down version of the logic in ApiForm. It validates the
+  // form data prior to submitting the request, and overrides the request host
+  // with the selected region's URL if one is provided.
+  const submitOrganizationCreate: OnSubmitCallback = useCallback((data, onSubmitSuccess, onSubmitError, _event, formModel) => {
+    if (!formModel.validateForm()) {
+      return;
+    }
+    const regionUrl = data.dataStorageLocation;
+
+    client.request("/organizations/", {
+      method:"POST",
+      data: removeDataStorageLocationFromFormData(data),
+      host: regionUrl,
+      success: onSubmitSuccess,
+      error: onSubmitError
+    });
+  }, [client]);
```

Comment:
I'm not 100% sure I need to wrap this in a `useCallback` hook, but figured I may as well cache this closure? Also not sure if the `useApi` hook + reference here is better than just initializing a new client object.

Reply:
You should use `useCallback` because you're in a component function. Without that you'll be re-assigning props to `Form` on each render. Another approach would be to move this function outside of the component and create the `Client` instance on your own. However, this will work as it is and is inline with how many other react components are structured.